docs/proposals/20230715-improve-yurthub-cache.md

```diff
@@ -0,0 +1,547 @@
+-   As the local storage data directory is: component/resource/namespace, according to the resource to write operations, even if there are more than one concurrent coprocessing, there will be no write conflict operation, so there is no need to lock processing, but also to improve the efficiency of writing disk
+
+#### Deployment Program
+
+-   Unify YurtHub deployment YAML files, instead of specifying whether to run in cloud mode or edge mode by entering a parameter, determine the YurtHub operation mode based on which node group the current node belongs to, and based on the type of node group. This way, YurtHub's final YAML file format will remain unified, and the yurt-static-set upgrade templates will be unified, making it easier to upgrade.
+
+#### Disk Storage Adjustment
+
+Problem 5 (list data storage conflict problem) is solved by adjusting the program interface as follows：
+
+-   Regarding list data storage, in the case of not changing the storage path, the use of concatenated storage, that is, the same resource for a variety of list results to seek a concatenation, and then stored, need to be processed, such as data query, and then based on the contents of the selector or the path, the program to do a good job of filtering and then return to the
+-   Regarding each re-list operation, you need to use the same list conditions to get the data from the local area, and then compare it with the data returned from the cloud: 1. if the cloud has it, but the side doesn't have it, create it; 2. if the cloud doesn't have it, but the side has it, delete it; 3. if the cloud and the side both have it, update it.
+
+
+### HF Resource Caching
+
+Considering the resource limitations of the edge device, the caching function inside the CacheManager should only cache the necessary high-frequency resources to reduce memory consumption. According to the conditions of the above query, we can conclude that the current YurtHub high-frequency query resources, so the current cache only caches the following resources, and will add them later if necessary.
+
+| component    | verb | resource  |
+| ------- | ---- | ----- |
+| kubelet | list | pods  |
+|         | get  | nodes |
```

Comment:
I think that `kubelet list pods` is not a high-frequency request, because kubelet will list pods only once when kubelet startup.

but `kubelet get lease` is a high-frequency request.